lib/jsdom/living/xhr/XMLHttpRequest-impl.js

```diff
@@ -301,7 +302,7 @@ class XMLHttpRequestImpl extends XMLHttpRequestEventTargetImpl {
         lastModified: new Date(getResponseHeader(this, "last-modified")),
         parsingMode: isHTML ? "html" : "xml",
         cookieJar: { setCookieSync: () => undefined, getCookieStringSync: () => "" },
-        encoding,
+        encoding: toCasedEncodingName(encoding),
```

Comment:
Rename the constructor privateData option to `Document` to `canonicalizedEncodingLabel` (which sets `_canonicalizedEncodingLabel`), and then we can just pass it in as-is.

Reply:
This was reverted